### PR TITLE
DSND-2138: Transform associated filing delta into request body

### PR DESCRIPTION
**Pull request by sthompsonCH**
## Describe the changes
This PR transforms an associated filing into a valid PUT request sent to the API.

### Related Jira tickets
[DSND-2138](https://companieshouse.atlassian.net/browse/DSND-2138)

### General
- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
_Add any testing hints or instructions here._

[DSND-2138]: https://companieshouse.atlassian.net/browse/DSND-2138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by kinpang-CH — APPROVED** [review]

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/kafka/ConsumerPositiveComprehensiveIT.java [hunk 1]
```diff
@@ -67,54 +67,56 @@ public void setup() {
...
-            "insolvency/AM15(Scot)", "insolvency/AM23(Scot)", "insolvency/NCOP", "insolvency/RM01(Scot)",
-            "mortgage/MR01_rule_4", "mortgage/MR01_rule_5", "mortgage/MR02_rule_3", "mortgage/MR02_rule_4",
-            "mortgage/MR03_rule_4",
-            "mortgage/MR04_rule_1", "mortgage/MR04_rule_2", "mortgage/MR04_rule_3", "mortgage/MR05_rule_1",
-            "mortgage/MR05_rule_2", "mortgage/MR05_rule_3", "mortgage/MR05_rule_4", "mortgage/MR05_rule_5",
-            "mortgage/MR05_rule_6", "mortgage/MR05_rule_7", "mortgage/MR06", "mortgage/MR07", "mortgage/MR08",
-            "mortgage/MR09", "mortgage/MR10",
-//            "mortgage/LLMR03", // Failing due to missing description value or invalid rule on line 4295
-            "mortgage/466(Scot)"
+            "associated_filing/associated_filing"
```
Comment (cgregoryCH) [inline src/test/java/uk/gov/companieshouse/filinghistory/consumer/kafka/ConsumerPositiveComprehensiveIT.java:70]: Do these all pass? If so can be uncommented

Comment (johncook-ch) [reply src/test/java/uk/gov/companieshouse/filinghistory/consumer/kafka/ConsumerPositiveComprehensiveIT.java:70]: "mortgage/LLMR03" is fixed in https://companieshouse.atlassian.net/browse/DSND-2353 which is yet to be merged

Comment (cgregoryCH) [reply src/test/java/uk/gov/companieshouse/filinghistory/consumer/kafka/ConsumerPositiveComprehensiveIT.java:70]: Can https://github.com/companieshouse/filing-history-delta-consumer/pull/56 me reviewed and merged first then please @sthompsonCH ?

**Review by cgregoryCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**